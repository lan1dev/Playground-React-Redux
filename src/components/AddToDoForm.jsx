import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import Input from './Input';

const propTypes = {
  addToDo: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  toDoValue: PropTypes.string.isRequired
};

const AddToDoForm = ({ addToDo, handleChange, toDoValue }) => (
  <form onSubmit={addToDo}>
    <Input className="todo-input" name="todo" onChange={handleChange} value={toDoValue} />
    <Button>Add</Button>
  </form>
);

AddToDoForm.propTypes = propTypes;
export default AddToDoForm;
