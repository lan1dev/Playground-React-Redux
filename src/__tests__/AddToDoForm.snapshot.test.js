import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddToDoForm from '../components/AddToDoForm';

describe('AddToDoForm', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<AddToDoForm {...props} />);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: undefined
    };
    wrapper = undefined;
  });

  describe('when component has rendered', () => {
    props = {
      addToDo: jest.fn(),
      handleChange: jest.fn(),
      addToDo: ''
    };
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
