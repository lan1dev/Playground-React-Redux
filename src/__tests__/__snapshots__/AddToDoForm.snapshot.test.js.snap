// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToDoForm when component has rendered Should match snapshot 1`] = `
<form>
  <styled.input
    className="todo-input"
    name="todo"
  />
  <styled.button>
    Add
  </styled.button>
</form>
`;
